src/main/scala/tech/sourced/gemini/Gemini.scala

```diff
@@ -15,14 +15,30 @@ import scala.io.Source
 
 class Gemini(session: SparkSession, keyspace: String = Gemini.defautKeyspace) {
 
-  def hash(reposPath: String): DataFrame = {
+  import session.implicits._
+
+  def hash(reposPath: String, limit: Int = 0): DataFrame = {
     if (session == null) {
       throw new UnsupportedOperationException("Hashing requires a SparkSession.")
     }
 
     val engine = Engine(session, reposPath)
 
-    val headRefs = engine.getRepositories
+    // engine.getRepositories.limit(n)...getFiles - doesn't work in engine now
+    // https://github.com/src-d/engine/issues/267
+    // use workaround with filter
+    if (limit <= 0) {
+       hashForRepos(engine.getRepositories)
+    } else {
+      val repoIds = engine.getRepositories.limit(limit).select($"id").collect().map {
+        r => r(0)
+      }
```

Comment:
I believe that in such case
```scala
...collect.map(_(0))
```
or even something like

```scala
.map(_.getAs[String]("id"))
```

that looks more concise should also work.

Reply:
yeah! thank you. I was looking for something like this. I mean `_(0)`. Not sure about `getAs`. Type and content are quite obvious in my opinion in this case.

Reply:
fixed